src/Apps/Order/Routes/Shipping2/AddressModal2.tsx

```diff
@@ -114,100 +106,131 @@ export const AddressModal: React.FC<Props> = ({
     setCreateUpdateError(null)
   }
 
+  const handleErrors = (
+    errors: ReadonlyArray<{ message: string }>,
+    formikHelpers
+  ) => {
+    if (!errors?.length) return
+
+    const userMessage: Record<string, string> | null =
+      SERVER_ERROR_MAP[errors[0].message]
+
+    if (userMessage) {
+      formikHelpers.setFieldError(userMessage.field, userMessage.message)
+    } else {
+      setCreateUpdateError(GENERIC_FAIL_MESSAGE)
+    }
+
+    formikHelpers?.setSubmitting(false)
+    logger.error(errors.map(error => error.message).join(", "))
+  }
+
   const handleDeleteAddress = async (addressID: string) => {
-    return deleteUserAddress(
-      relayEnvironment!,
-      addressID,
-      onDeleteAddress,
-      onError
-    )
+    try {
+      return deleteSavedAddress({
+        variables: {
+          input: { userAddressID: addressID },
+        },
+      })
+    } catch (error) {
+      logger.error(error)
+    }
   }
 
+  const handleMutationPayload = (
+    payload:
+      | useUpdateSavedAddressMutation$data["updateUserAddress"]
+      | useCreateSavedAddressMutation$data["createUserAddress"]
+  ):
+    | { data: SavedAddressType; errors: null }
+    | { data: null; errors: ReadonlyArray<{ message: string }> } => {
+    const addressOrErrors = payload?.userAddressOrErrors
+    const errors = addressOrErrors?.errors
+    if (errors?.length) {
+      return { errors, data: null }
+    }
+    return { errors: null, data: addressOrErrors as SavedAddressType }
+  }
+
+  const handleSubmit = async (
+    values: SavedAddressType,
+    helpers: FormikHelpers<SavedAddressType>
+  ) => {
+    const addressInput = addressWithFallbackValues(values)
+    let operation: () => Promise<ReturnType<typeof handleMutationPayload>>
+    try {
+      if (modalAction.type === "createUserAddress") {
+        operation = () => {
+          return createSavedAddress({
+            variables: {
+              input: { attributes: addressInput },
+            },
+          }).then(result => {
+            return handleMutationPayload(result.createUserAddress)
+          })
+        }
+      } else {
+        operation = () =>
+          updateSavedAddress({
+            variables: {
+              input: {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                userAddressID: initialAddress.internalID!,
+                attributes: addressInput,
+              },
+            },
+          }).then(result => handleMutationPayload(result.updateUserAddress))
+      }
+      const { data, errors } = await operation()
+      if (errors) {
+        handleErrors(errors, helpers)
+        return
+      }
+      const savedAddressID = data?.internalID
```

Comment:
This is #superminor, but just something I frequently notice when reviewing PRs and something to be aware of for future (other devs) readability  -- there's rarely ever any space between statements. Like compare the below:

```tsx
const addressInput = addressWithFallbackValues(values)
let operation: () => Promise<ReturnType<typeof handleMutationPayload>>
try {
  ...
catch () { 
  ...
}
const data = await operation()
if (errors) {
 ... 
}
const savedAddressID = foo 
```

vs 

```tsx 
const addressInput = addressWithFallbackValues(values)
let operation: () => Promise<ReturnType<typeof handleMutationPayload>>

try {
  ...
catch () { 
  ...
}

const data = await operation()

if (errors) {
 ... 
}

const savedAddressID = foo 
```

In the latter, a dev reading yr code can _scan_, and notice when the scenery changes. New things are signified by significant white space. With everything constantly (or arbitrarily) squished together, folks reading code in the future are forced to squint. 

Its a non-blocking thing, but legibility is really important when working with non-trivial code on a larger team. (This also goes for PR descriptions where paragraph breaks are frequently omitted between key sections, making things feel needlessly sloppy.)

Reply:
yeah I completely agree with this and am normally way more nitpicky about it 👍

Reply:
Addressed it in this place, will keep my eyes out for other big functions that could benefit from it in the future